node/pkg/accounting/accounting.go

```diff
@@ -0,0 +1,353 @@
+		}
+
+		return true, nil
+	}
+
+	// We only care about transfers.
+	if !vaa.IsTransfer(msg.Payload) {
+		acct.logger.Info("acct: ignoring vaa because it is not a transfer", zap.String("msgID", msgId))
+		return true, nil
+	}
+
+	digest := msg.CreateDigest()
+
+	acct.pendingTransfersLock.Lock()
+	defer acct.pendingTransfersLock.Unlock()
+
+	// If this is already pending, don't send it again.
+	if oldEntry, exists := acct.pendingTransfers[msgId]; exists {
+		if oldEntry.digest != digest {
+			digestMismatches.Inc()
+			acct.logger.Error("acct: digest in pending transfer has changed, dropping it",
+				zap.String("msgID", msgId),
+				zap.String("oldDigest", oldEntry.digest),
+				zap.String("newDigest", digest),
+			)
+		} else {
+			acct.logger.Info("acct: blocking transfer because it is already outstanding", zap.String("msgID", msgId))
+		}
+		return false, nil
+	}
+
+	// Add it to the pending map and the database.
+	if err := acct.addPendingTransfer(msgId, msg, digest); err != nil {
+		acct.logger.Error("acct: failed to persist pending transfer, blocking publishing", zap.String("msgID", msgId), zap.Error(err))
+		return false, err
+	}
+
+	// This transaction may take a while. Pass it off to the worker so we don't block the processor.
+	if acct.env != GoTestMode {
+		acct.logger.Info("acct: submitting transfer to accounting for approval", zap.String("msgID", msgId), zap.Bool("canPublish", !acct.enforceFlag))
+		acct.submitObservation(msg)
+	}
+
+	// If we are not enforcing accounting, the event can be published. Otherwise we have to wait to hear back from the contract.
+	return !acct.enforceFlag, nil
+}
+
+// AuditPending audits the set of pending transfers for any that have been in the pending state too long. This is called from the processor loop
+// each timer interval. Any transfers that have been in the pending state too long will be resubmitted. Any that has been retried too many times
+// will be logged and dropped.
+func (acct *Accounting) AuditPendingTransfers() {
+	acct.logger.Debug("acct: in AuditPendingTransfers")
+	acct.pendingTransfersLock.Lock()
+	defer acct.pendingTransfersLock.Unlock()
+
+	if len(acct.pendingTransfers) == 0 {
+		acct.logger.Debug("acct: leaving AuditPendingTransfers, no pending transfers")
+		return
+	}
+
+	for msgId, pe := range acct.pendingTransfers {
+		acct.logger.Debug("acct: evaluating pending transfer", zap.String("msgID", msgId), zap.Stringer("updTime", pe.updTime))
+		if time.Since(pe.updTime) > auditInterval {
+			pe.retryCount += 1
+			if pe.retryCount > maxRetries {
+				acct.logger.Error("acct: stuck pending transfer has reached the retry limit, dropping it", zap.String("msgId", msgId))
+				acct.deletePendingTransfer(msgId)
+				continue
+			}
+
+			acct.logger.Error("acct: resubmitting pending transfer",
+				zap.String("msgId", msgId),
+				zap.Stringer("lastUpdateTime", pe.updTime),
+				zap.Int("retryCount", pe.retryCount),
+			)
+
+			pe.updTime = time.Now()
+			acct.submitObservation(pe.msg)
```

Comment:
This will break all token transfers the moment 12 transactions are stuck without quorum because the guardian will try to resubmit each of them in a blocking way every minute with 5s block times which fully exhausts the sender. We probably need to query and do some kind of exponential backoff (#2002)

Reply:
Yes, issue #2002 will address the audit redesign.

Reply:
Ok as long as we keep this in mind as a pre-release item we're good.